File: BAEKJOON/go/10000/10400/10448.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)
var wr = bufio.NewWriter(os.Stdout)

func nextInt() int {
	sc.Scan()
	text := sc.Text()
	v, _ := strconv.Atoi(text)
	return v
}

func findTriangulars(triangularNumbers []int, K int) int {
	for _, x := range triangularNumbers {
		for _, y := range triangularNumbers {
			for _, z := range triangularNumbers {
				if x+y+z == K {
					return 1
				}
			}
		}
	}
	return 0
}

func main() {
	sc.Split(bufio.ScanWords)
	defer wr.Flush()

	var triangularNumbers []int
	triangularNumbers = append(triangularNumbers, 1)
	for i := 2; triangularNumbers[i-2] < 1000; i++ {
		triangularNumbers = append(triangularNumbers, triangularNumbers[i-2]+i)
	}
	testCase := nextInt()
	for i := 0; i < testCase; i++ {
		K := nextInt()
		fmt.Fprintln(wr, findTriangulars(triangularNumbers, K))
	}
}

File: BAEKJOON/Python/10000/10400/10448.py
"""
Title : 유레카 이론
Link : https://www.acmicpc.net/problem/10448
"""

from sys import stdin

input = stdin.readline
II = lambda: int(input())


def find_triangulars(triangular_numbers: list, K: int) -> int:
    for x in triangular_numbers:
        for y in triangular_numbers:
            for z in triangular_numbers:
                if x + y + z == K:
                    return 1
    return 0


if __name__ == "__main__":
    triangular_numbers = [1]
    for i in range(2, 1000):
        triangular_numbers.append(triangular_numbers[-1] + i)
        if triangular_numbers[-1] > 1000:
            break
    for _ in range(II()):
        print(find_triangulars(triangular_numbers, II()))
